test/dependency-manager-adapters/npm-adapter-test.js

```diff
@@ -59,6 +59,43 @@ describe('npmAdapter', () => {
     });
   });
 
+  describe('#changeToDependencySet', () => {
+    it('restores the backup before changing to the new dependency set', () => {
```

Comment:
I am fairly opposed to this 😸, setting up a new scenario to run should not itself require resetting to the backup first. This resetting seems to be masking another fundamental problem..

Reply:
Yea, I agree. It does feel a bit strange to reset to the backup first, instead, it should restore the backup at the end of each scenario. Let me look into our particular use case more closely and will get back to you.